### PR TITLE
Configure webpack production build to avoid using devtools that rely on eval

### PR DESCRIPTION
**Pull request by gabrielbusta**
Looks like Neutrino was doing this for us.
This should take care of the eval calls in the transpiled production code.

### CODE REVIEW

File: ui/webpack.config.js [hunk 1]
```diff
@@ -17,241 +17,243 @@ const eslintConfig = {
...
-    globalObject: 'this',
-  },
-  resolve: {
-    alias: {
-      'react-native': 'react-native-web',
-      'react-dom': '@hot-loader/react-dom',
+module.exports = env => {
+  return {
+    mode: env,
+    devtool: env === 'production' ? false : 'cheap-module-eval-source-map',
```
Comment (gabrielBusta) [inline ui/webpack.config.js:23]: Don't use `'cheap-module-eval-source-map'` in production builds.

File: ui/webpack.config.js [hunk 2]
```diff
@@ -17,241 +17,243 @@ const eslintConfig = {
...
-    path: `${__dirname}/build`,
-    publicPath: '/',
-    filename: 'assets/[name].js',
-    globalObject: 'this',
-  },
-  resolve: {
-    alias: {
-      'react-native': 'react-native-web',
-      'react-dom': '@hot-loader/react-dom',
+module.exports = env => {
```
Comment (gabrielBusta) [inline ui/webpack.config.js:20]: There's a lot of noise around here because the indentation changed.

**Review by bhearsum — COMMENTED** [review]
I don't really understand all of the changes to `webpack.config.js` -- is this a generated file?

File: ui/webpack.config.js [hunk 3]
```diff
@@ -17,241 +17,243 @@ const eslintConfig = {
...
-                        'last 2 Edge versions',
-                        'last 2 Opera versions',
-                        'last 2 Safari versions',
-                        'last 2 iOS versions',
-                      ],
+      host: process.env.HOST || DEFAULT_HOST,
+      https: true,
+      headers: {
+        'Content-Security-Policy':
+          "default-src 'none'; script-src 'self' 'unsafe-eval'; img-src 'self' https://*.gravatar.com https://*.githubusercontent.com https://i1.wp.com; style-src 'self' 'unsafe-inline' https://fonts.googleapis.com; font-src https://fonts.gstatic.com; connect-src https://localhost:8010 'self' https://balrog-localdev.auth0.com https://www.googleapis.com/; frame-src https://balrog-localdev.auth0.com; frame-ancestors 'none'; base-uri 'none'; form-action 'none'",
```
Comment (gabrielBusta) [inline ui/webpack.config.js:74]: I see `'unsafe-eval'` here. I wonder why this is an issue now. It is what Neutrino was generating...

File: ui/webpack.config.js [hunk 4]
```diff
@@ -17,241 +17,243 @@ const eslintConfig = {
...
+                          'last 2 iOS versions',
+                        ],
+                      },
                     },
-                  },
+                  ],
+                  [
+                    '@babel/preset-react',
+                    {
+                      development: env === 'development',
```
Comment (gabrielBusta) [inline ui/webpack.config.js:128]: Don't set `development` to true in production builds.

File: ui/webpack.config.js [hunk 5]
```diff
@@ -17,241 +17,243 @@ const eslintConfig = {
...
                 ],
-                [
-                  '@babel/preset-react',
-                  {
-                    development: true,
-                    useSpread: true,
-                  },
+                plugins: [
+                  '@babel/plugin-syntax-dynamic-import',
+                  ...(env === 'development' ? ['react-hot-loader/babel'] : []),
```
Comment (gabrielBusta) [inline ui/webpack.config.js:135]: Don't use `react-hot-loader/babel` in production builds.

**Comment by bhearsum** [comment]
> I don't really understand all of the changes to `webpack.config.js` -- is this a generated file?

Oh - nevermind. Hiding whitespace gets rid of most of it.

File: ui/webpack.config.js (hunk 1 above)
Comment (bhearsum) [reply ui/webpack.config.js:23]: This is probably worth a comment in the code explaining why not.

File: ui/webpack.config.js (hunk 5 above)
Comment (bhearsum) [reply ui/webpack.config.js:135]: Same thing here.

**Comment by gabrielBusta** [comment]
Yes. The reason for this is that we are now reading the env and configuring the build based on that. Because the export is now a function instead of a plain object the indentation changed and it looks like more changes than it is.

**Review by bhearsum — APPROVED** [review]

**Pull request merged**